simulator/func_sim/syscall/mars_calls.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * mars_calls.h - implementation of MARS syscalls
+ * @author Vyacheslav Kompan kompan.vo@phystech.edu
+ * Copyright 2018 MIPT-MIPS
+ */
+
+#ifndef MARS_CALLS_H
+#define MARS_CALLS_H
+
+/* Simulator modules */
+#include <func_sim/rf/rf.h>
+#include <infra/types.h>
+#include <syscall/syscall.h>
+/* Generic C & C++ */
+#include <cassert>
+
+template <typename ISA>
+class MARSCalls : public Syscall<ISA> {
+    using RegisterUInt = typename ISA::RegisterUInt;
+    using RegisterInt = sign_t<RegisterUInt>;
+
+    RF<ISA> *rf;
+
+    std::istream& instream;
+    std::ostream& outstream;
+
+    void print_integer();
+    void read_integer();
+    void exit() { throw TerminatedByExit(); }
```

Comment:
Again, if you have a pointer to Simulator instance, you may call a `halt()` method.

Reply:
Do I understand correctly the exception is not handled and we have abnormal (exit_code == 2) termination instead of returning 0?
Also, performance simulation prints some statistics — we do not want to skip it.

Reply:
Yes, I didn't notice it. We may catch it in `FuncSim::run()`. Another possibility, which I think is better, is to return a boolean which stops `FuncSim::run()` instead of exception. Same for PerfSim.